telluric/georaster.py

```diff
@@ -1638,6 +1640,55 @@ def _calculate_new_affine(self, window, blockxsize=256, blockysize=256):
         new_affine = new_affine * Affine.scale(x_scale, y_scale)
         return new_affine
 
+    def colorize(self, colormap, band_name=None, vmin=None, vmax=None):
+        """Apply a colormap on a selected band.
+
+        colormap list: https://matplotlib.org/examples/color/colormaps_reference.html
+
+        Parameters
+        ----------
+        colormap : str
+        Colormap name from this list https://matplotlib.org/examples/color/colormaps_reference.html
+
+        band_name : str, optional
+        Name of band to colorize, if none the first band will be used
+
+        vmin, vmax : int, optional
+        minimum and maximum range for normalizing array values, if None actual raster values will be used
+
+        Returns
+        -------
+        GeoRaster2
+        """
+        vmin = vmin or min(self.min())
+        vmax = vmax or max(self.max())
+        cmap = plt.get_cmap(colormap)
+
+        band_index = 0
+        if band_name is None:
+            if self.num_bands > 1:
+                warnings.warn("Using the first band to colorize the raster", GeoRaster2Warning)
+        else:
+            band_index = self.band_names.index(band_name)
+
+        normalized = np.divide(self.image.data[band_index, :, :] - vmin, vmax - vmin)
+        image_data = cmap(normalized)
```

Comment:
What does this part of the code do? `cmap(normalized)` returns an array with (x, y, bands) I guess, then we have to use only the first three bands? Could you point out to some matplotlib documentation on how this works?